src/Features/Core/Portable/CodeRefactorings/AddMissingImports/AbstractAddMissingImportsFeatureService.cs

```diff
@@ -126,22 +124,33 @@ private async Task<Document> ApplyFixesAsync(Document document, ImmutableArray<A
             }
```

Comment:
i'm still very concerned about blinding unioning text changed.  i think we should ensure these text changes do not conflict with other changes.

Reply:
I am not certain what pitfall I am trying to avoid. I was thinking about what the TextChanges would look like. They would all be insertions of the form "using baz;\r\n". Can you describe a scenario where applying these text changes would cause problems?

Reply:
That's not at all guaranteed.  We don't control the text-differencing service.  For all we know they may end up returning unexpected (but legal) data.  For example, if you add "using System.Collections.Generic;" and the code was:

```c#
using System;
using Generic;
```

They might represent that as multiple inserts (one before the existing 'Generic' one after it).

You're *hoping* that they will do a diff that always ends up with non-overlapping, obvious chunks here.  But you may not get that.

--

Note: to help out here, we probably want to pass along a parameter that changes this:

```c#
            var differenceOptions = new StringDifferenceOptions()
            {
                DifferenceType = StringDifferenceTypes.Word
            };
```

Right now we're doing word diffing, which is far more likely to run into text changes that have problems.  Based on what we know of usings/imports, it would be great if we could *parameterize* this so we can pass StringDifferenceTypes.Line along instead.  that way the editor far more often gives us the diff we expect.

(you'll still need to validate that the results you get back don't overlap).

Reply:
> If I added an optional parameter to the interface called something like preferredDifferenceOptions and allowed the caller to pass in their StringDifferenceOptions would it be a problem if not all IDocumentTextDifferencingServices respected them?

Yeah, i think this would be ok.  It can just be documented as a best-effort parameter.  

> To check for overlap, would checking that the text change is an insertion (meaning the replacement span is empty) suffice or were you thinking something different?

That's cute.  I think it would work great! :)

Reply:
Passing in the StringDifferenceOptiosn itself didn't work out because of which layer each of these classes are in. Instead I added an enum that matches StringDifferenceTypes, pass that in, and create Options from it.